pkg/typesystem/typesystem.go

```diff
@@ -581,24 +580,27 @@ func validateRelationTypeRestrictions(model *openfgapb.AuthorizationModel) error
 				relatedObjectType := related.GetType()
 				relatedRelation := related.GetRelation()
 
-				if _, err := t.GetRelations(relatedObjectType); err != nil {
+				if _, err := typeSystem.GetRelations(relatedObjectType); err != nil {
 					return InvalidRelationTypeError(objectType, name, relatedObjectType, relatedRelation)
 				}
 
-				if relatedRelation != "" {
-					if _, err := t.GetRelation(relatedObjectType, relatedRelation); err != nil {
-						return InvalidRelationTypeError(objectType, name, relatedObjectType, relatedRelation)
-					}
-
-					// you cannot specify a userset if the relation is being used in a `x from y` definition (in the `y` part)
-					for _, arrayOfTtus := range allTupleToUsersetDefinitions[objectType] {
+				if related.GetRelationOrWildcard() != nil {
+					// you cannot specify a userset or wildcard if the relation is being used in a `x from y` definition (in the `y` part)
+					for _, arrayOfTtus := range typeSystem.GetAllTupleToUsersetsDefinitions()[objectType] {
```

Comment:
Can we simplify this code by adding a helper method that we can use here and elsewhere for 'tupleset' relations.

It'd be nice if our Typesystem has a method

`IsTuplesetRelation(objectType, relation string) (bool, error)`

Then wherever we're using `GetAllTupleToUsersetDefinitions` and indexing to a specific one we can just call `IsTuplesetRelation(...)`.

This code would simplify to

```
if related.GetRelationOrWildcard() != nil {
    isTupleset, err := IsTuplesetRelation(...)
    if isTupleset {
        return fmt.Errorf("invalid type type restriction for tupleset relation ...")
    }
}
```

Reply:
Good idea. I think you added this function in your validation PR.

Done. Thanks!